File: ProjectEuler/Problem_1/1-Multiples_of_3_and_5.hs
-- | Project Euler
-- 1-Find the sum of all the multiples of 3 or 5 below 1000.
-- Optimized
sum [3,6..999] + sum [5,10..999] - sum [15,30..999]
-- Brute force
sum  [x | x <- [0..999], x `mod` 3 == 0 || x `mod` 5 == 0]

File: ProjectEuler/Problem_1/1-Multiples_of_3_and_5.r
# Project Euler
# Problem 1 1-Find the sum of all the multiples of 3 or 5 below 1000.
# Optimized
sumby <- function(a) sum(seq(a, 999, by=a))
sumby(3) + sumby(5) - sumby(15)
# 233168
# Brute force
l <- 1:999
sum(l[l %% 3 == 0 | l %% 5 == 0])
# 233168